File: valSubsequence.js
function validateSubsequence(array, subsequence) {
    let indexArr = 0;
    let indexSeq = 0;
    while (indexArr < array.lenght && indexSeq < subsequence.lenght) {
        if (array[indexArr] === subsequence[indexSeq]) indexSeq++;
        indexArr++;
    }
    return indexSeq === subsequence.lenght;
}

exports.validateSubsequence = validateSubsequence;

// validates if second array is subsequence of the first array

File: valSubsequence.py
def isValidSubsequence(array, sequence):
    """
    validates if second array is subsequence of the first array
    """
    index_arr = 0
    index_seq = 0

    while index_arr < len(array) and index_seq < len(sequence):
        if array[index_arr] == sequence[index_seq]:
            index_seq += 1
        index_arr += 1
    return index_seq == len(sequence)
    